blog/2023-04-23-json.md

```diff
@@ -0,0 +1,172 @@
+---
+title: "Json Takes Flight"
+description: Why Wing has a built-in Json primitive
+authors: 
+  - hasanaburayyan
+tags: [cloud-oriented programming, winglang, resources, preflight, inflight, iac, security]
+hide_table_of_contents: true
+---
+
+Picture this: you're in the zone, crafting some spectacular code like the rock star cloud developer you are. You know that to truly harness the power of the cloud ecosystem, your system needs to be distributed. After an all-night coding spree, you've created two nifty services lovingly named service1 and service2. Now, service1 must produce a message and send it to service2. So, it's time to choose a message format standard. Given that it's 2023, XML is off the table, making JSON the clear winner.
+
+At this point, something odd happens in your development journey. You pause to figure out how to use JSON in your chosen programming language, only to discover that you need to import a library. Sure, it's manageable, but now you have to consider the implications of using this JSON library and compare it to others. Of course, you'll make the right choice, you are a rock star, after all. However, this experience leaves you wondering: with JSON being so common, shouldn't it be built into any language designed for distributed computing?
+
+## Communication Is Key
+
+When designing a good distributed system, several key aspects come to mind, such as fault tolerance, scalability, security, high availability, and interoperability. In this section, we'll focus on interoperability—the ability of a system to work seamlessly with other systems. This means that communication between systems must adhere to not only standardized protocols but also data formats.
```

Comment:
Don't know if it's just me, but I had to read the part with the "interoperability—the ability .." to understand it
```suggestion
When designing a good distributed system, several key aspects come to mind, such as fault tolerance, scalability, security, high availability, and interoperability. In this section, we'll focus on interoperability — the ability of a system to work seamlessly with other systems. This means that communication between systems must adhere to not only standardized protocols but also data formats.
```

Another option would be to put the explanation in parenthesis